internal/ext/common.go

```diff
@@ -13,14 +15,15 @@ type Document struct {
 }
 
 type Flag struct {
-	Key         string     `yaml:"key,omitempty" json:"key,omitempty"`
-	Name        string     `yaml:"name,omitempty" json:"name,omitempty"`
-	Type        string     `yaml:"type,omitempty" json:"type,omitempty"`
-	Description string     `yaml:"description,omitempty" json:"description,omitempty"`
-	Enabled     bool       `yaml:"enabled" json:"enabled"`
-	Variants    []*Variant `yaml:"variants,omitempty" json:"variants,omitempty"`
-	Rules       []*Rule    `yaml:"rules,omitempty" json:"rules,omitempty"`
-	Rollouts    []*Rollout `yaml:"rollouts,omitempty" json:"rollouts,omitempty"`
+	Key         string           `yaml:"key,omitempty" json:"key,omitempty"`
+	Name        string           `yaml:"name,omitempty" json:"name,omitempty"`
+	Type        string           `yaml:"type,omitempty" json:"type,omitempty"`
+	Description string           `yaml:"description,omitempty" json:"description,omitempty"`
+	Enabled     bool             `yaml:"enabled" json:"enabled"`
+	Metadata    *structpb.Struct `yaml:"metadata,omitempty" json:"metadata,omitempty"`
```

Comment:
```suggestion
	Metadata   map[string]string `yaml:"metadata,omitempty" json:"metadata,omitempty"`
```

To follow on from @markphelps suggestion. If we just make this a `map[string]string` here we can use `Metadata.AsMap()` when populating this instead.